src/auth0/cache/crypto.rs

```diff
@@ -1,20 +1,29 @@
-use aes::Aes256 as Aes256Alg;
-use block_modes::block_padding::Pkcs7;
-use block_modes::{BlockMode, Cbc};
+use aes::{
+    cipher::{block_padding::Pkcs7, BlockDecryptMut, BlockEncryptMut, KeyIvInit},
+    Aes256,
+};
+use cbc::{Decryptor, Encryptor};
 use serde::{Deserialize, Serialize};
 
 use crate::auth0::errors::Auth0Error;
 
-type Aes256 = Cbc<Aes256Alg, Pkcs7>;
+type Aes256Enc = Encryptor<Aes256>;
+type Aes256Dec = Decryptor<Aes256>;
 
 const IV: &str = "301a9e39735f4646";
 
 pub fn encrypt<T: Serialize>(value_ref: &T, token_encryption_key_str: &str) -> Result<Vec<u8>, Auth0Error> {
     let json: String = serde_json::to_string(value_ref)?;
+
     // `unwrap` here is fine because `IV` is set here and the only error returned is: `InvalidKeyIvLength`
     // and this must never happen
-    let cipher: Aes256 = Aes256::new_from_slices(token_encryption_key_str.as_bytes(), IV.as_bytes()).unwrap();
-    Ok(cipher.encrypt_vec(json.as_bytes()))
+    let mut buf = vec![0u8; json.len()];
```

Comment:
This is probably not right: in the [example](https://docs.rs/cbc/0.1.2/cbc/#example) `buf` is a bit larger than the plaintext and it's stated that
> buffer must be big enough for padded plaintext
my problem is that I don't know how big the padded plaintext will be